services/type-utils/src/main/java/datawave/data/type/util/PhoneNumber.java

```diff
@@ -382,6 +382,13 @@ public boolean equals(Object o) {
         }
     }
     
+    @Override
+    public int hashCode() {
+        int result = originalPhoneNumber != null ? originalPhoneNumber.hashCode() : 0;
```

Comment:
The equals method compares the result of toNormalizedString, however we are generating a hash code based on the unnormalized value.  If two different unnormalized values normalize to the same thing, then equals would return true but we'd compute different hash codes.

Reply:
Ok, not good at reading today. I see the normalized number is added to the hash below. Still, it seems off to be that we're using different fields (both unnormalized and normalized) for the hashCode vs equals (only the normalized number is used).

Reply:
No, you're right, should've just been based on the normalized value.  I think I was starting to suffer from repetitive-stress injury at this point, from clicking `Code --> Generate... --> .... ` in Intellij ;)

Reply:
The equals impl is somewhat suspect as well. It doesn't guard against NPE, which could easily occur if null was passed into constructor.

Reply:
Updated both equals and hashCode in latest commit